### PR TITLE
feat: shield commands for all APIs

### PR DESCRIPTION
**Pull request by ishanarya0**
Command groups to be added:

- [x] User
- [x] Group
- [x] Roles
- [x] Organizations
- [x] Project
- [x] Actions
- [x] Namespaces
- [x] Policies

### CODE REVIEW

**Comment by ishanarya0** [comment]
Namespace commands:

```
$ shield namespace create --file=<namespace-body>
$ shield namespace update <namespace-id> --file=<namespace-body>
$ shield namespace get <namespace-id>
$ shield namespace list
```

**Comment by ishanarya0** [comment]
User commands:

```
$ shield user create --file=<user-body>
$ shield user update <user-id> --file=<user-body>
$ shield user get <user-id>
$ shield user list
```

**Comment by ishanarya0** [comment]
Organization commands:

```
$ shield organization create --file=<organization-body>
$ shield organization update <organization-id> --file=<organization-body>
$ shield organization get <organization-id>
$ shield organization list
// $ shield organization admadd <organization-id> -file=<add-organization-admin-body>
// $ shield organization admremove <organization-id> --user=<user-id>
// $ shield organization admlist <organization-id>
```

Note: ADD, REMOVE, and LIST commands for organization admins have been commented on since their respective functions need to be implemented in shield service.

**Comment by ishanarya0** [comment]
Group commands:

```
$ shield group create --file=<group-body>
$ shield group update <group-id> --file=<group-body>
$ shield group get <group-id>
$ shield group list
```

**Comment by ishanarya0** [comment]
Project commands:

```
$ shield project create --file=<project-body>
$ shield project update <project-id> --file=<project-body>
$ shield project get <project-id>
$ shield project list
```

**Comment by ishanarya0** [comment]
Role commands:

```
$ shield role role --file=<role-body>
$ shield role update <role-id> --file=<role-body>
$ shield role get <role-id>
$ shield role list
```

**Comment by ishanarya0** [comment]
Action commands:

```
$ shield action role --file=<action-body>
$ shield action update <action-id> --file=<action-body>
$ shield action get <action-id>
$ shield action list
```

File: cmd/action.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+	return cmd
+}
+
+func createActionCommand(logger log.Logger, appConfig *config.Shield) *cli.Command {
+	var filePath, header string
+
+	cmd := &cli.Command{
+		Use:   "create",
+		Short: "create all actions",
```
Comment (ravisuhag) [inline cmd/action.go:50]: @ishanarya0 Description would be "Create an action". First letter cap. We need to follow similar convention across all descriptions. Check Guardian for ref.

File: cmd/action.go [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+	return cmd
+}
+
+func updateActionCommand(logger log.Logger, appConfig *config.Shield) *cli.Command {
+	var filePath string
+
+	cmd := &cli.Command{
+		Use:   "update",
+		Short: "update all actions",
```
Comment (ravisuhag) [inline cmd/action.go:108]: "Update an action"

File: cmd/action.go [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+	cmd.Flags().StringVarP(&filePath, "file", "f", "", "Path to the action body file")
+	cmd.MarkFlagRequired("file")
+
+	return cmd
+}
+
+func getActionCommand(logger log.Logger, appConfig *config.Shield) *cli.Command {
+	cmd := &cli.Command{
+		Use:   "get",
+		Short: "get an action",
```
Comment (ravisuhag) [inline cmd/action.go:162]: View an action

File: cmd/action.go [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+	}
+
+	return cmd
+}
+
+func listActionCommand(logger log.Logger, appConfig *config.Shield) *cli.Command {
+
+	cmd := &cli.Command{
+		Use:   "list",
+		Short: "list all actions",
```
Comment (ravisuhag) [inline cmd/action.go:215]: List all actions

**Comment by ravisuhag** [comment]
> Namespace commands:
> 
> ```
> $ shield namespace create --file=<namespace-body>
> $ shield namespace update <namespace-id> --file=<namespace-body>
> $ shield namespace get <namespace-id>
> $ shield namespace list
> ```

@ishanarya0 Let's rename all `update` to `edit`. And all `get` to `view`. Also are we adding delete command?

File: cmd/namespace.go [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+	"github.com/MakeNowJust/heredoc"
+	"github.com/odpf/salt/log"
+	"github.com/odpf/salt/printer"
+	"github.com/odpf/shield/config"
+	shieldv1beta1 "github.com/odpf/shield/proto/v1beta1"
+	cli "github.com/spf13/cobra"
+)
+
+func NamespaceCommand(logger log.Logger, appConfig *config.Shield) *cli.Command {
+
```
Comment (ravisuhag) [inline cmd/namespace.go:18]: We can remove empty line.

**Comment by ishanarya0** [comment]
> > Namespace commands:
> > ```
> > $ shield namespace create --file=<namespace-body>
> > $ shield namespace update <namespace-id> --file=<namespace-body>
> > $ shield namespace get <namespace-id>
> > $ shield namespace list
> > ```
> 
> @ishanarya0 Let's rename all `update` to `edit`. And all `get` to `view`. Also are we adding delete command?

I have already made the changes in all the commands, in my last commit. Missed out to update it in these comments.

I couldn't find an API to support the delete operation. If there is one, I can incorporate it as well.

File: cmd/action.go [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+	cmd.MarkFlagRequired("header")
+
+	return cmd
+}
+
+func updateActionCommand(logger log.Logger, appConfig *config.Shield) *cli.Command {
+	var filePath string
+
+	cmd := &cli.Command{
+		Use:   "update",
```
Comment (ravisuhag) [inline cmd/action.go:106]: @ishanarya0 this update command should be renamed to `edit`. Same for the rest.

File: cmd/action.go [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+
+	cmd.Flags().StringVarP(&filePath, "file", "f", "", "Path to the action body file")
+	cmd.MarkFlagRequired("file")
+
+	return cmd
+}
+
+func getActionCommand(logger log.Logger, appConfig *config.Shield) *cli.Command {
+	cmd := &cli.Command{
+		Use:   "get",
```
Comment (ravisuhag) [inline cmd/action.go:160]: command name should also be renamed to `view` instead of get.

**Comment by ravisuhag** [comment]
> > > Namespace commands:
> > > ```
> > > $ shield namespace create --file=<namespace-body>
> > > $ shield namespace update <namespace-id> --file=<namespace-body>
> > > $ shield namespace get <namespace-id>
> > > $ shield namespace list
> > > ```
> > 
> > 
> > @ishanarya0 Let's rename all `update` to `edit`. And all `get` to `view`. Also are we adding delete command?
> 
> I have already made the changes in all the commands, in my last commit. Missed out to update it in these comments.
> 
> I couldn't find an API to support the delete operation. If there is one, I can incorporate it as well.

@ishanarya0 command name should be changed as well.

File: cmd/action.go (hunk 7 above)
Comment (ishanarya0) [reply cmd/action.go:160]: oh, okay my bad

**Comment by ravisuhag** [comment]
@ishanarya0 Please check tests.

**Pull request merged**